README.md
[<img src="https://api.travis-ci.org/chaica/retweet.svg?branch=master">](https://travis-ci.org/chaica/retweet)
### Retweet 

Retweet automatically retweets tweets from a Twitter user. Read the documentation in docs/
or [read it online](https://retweet.readthedocs.org/en/latest/).

### Quick Install

* Install Retweet from PyPI

        # pip3 install retweet

* Install Retweet from sources    
  *(see the installation guide for full details)
  [Installation Guide](http://retweet.readthedocs.org/en/latest/install.html)*
  

        # tar zxvf retweet-0.9.tar.gz
        # cd retweet
        # python3.4 setup.py install
        # # or
        # python3.4 setup.py install --install-scripts=/usr/bin

### Use Retweet

* Create or modify retweet.ini file in order to configure Retweet:

        [twitter]
        screen_name_of_the_user_to_retweet=journalduhacker
        consumer_key=ml9jaiBnf3pmU9uIrKNIxAr3v
        consumer_secret=8Cmljklzerkhfer4hlj3ljl2hfvc123rezrfsdctpokaelzerp
        access_token=213416590-jgJnrJG5gz132nzerl5zerwi0ahmnwkfJFN9nr3j
        access_token_secret=3janlPMqDKlunJ4Hnr90k2bnfk3jfnwkFjeriFZERj32Z

        [retweet]
        retweets=0
        waitminsecs=60
        waitmaxsecs=600
        do_not_retweet_hashtags=dnr,
        only_if_hashtags=python,
        like=true

        [sqlite]
        sqlitepath=/var/lib/retweet/retweet.db

* Launch Retweet

        $ retweet /path/to/retweet.ini

### Authors

Carl Chenet <chaica@ohmytux.com>

### License

This software comes under the terms of the GPLv3+. See the LICENSE file for the complete text of the license.


CHANGELOG
 Change Log
All notable changes to this project will be documented in this file.

## [0.9] - 2016-09-02
### Added
- like parameter to like all retweeted tweets.
- --dry-run command line option for testing purposes.
- --version command line option to display current version.

### Changed
- split [main] section in configuration file to [twitter] and [retweet] sections.

## [0.8] - 2016-01-10
### Added
- older_than parameter to retweet only if the tweet is older than a given number of minutes.
- younger_than parameter to retweet only if the tweet is younger than a given number of minutes.
- -l or --limit command line option to retrieve only the given number of statuses from Twitter.

## [0.7] - 2015-12-22
### Changed
- bug fix

## [0.6] - 2015-11-19
### Added
- only_if_hashtags parameter to retweet only tweets containing one of the hashtags of this list.

### Changed
- rename do_not_retweet_hashes parameter to do_not_retweet_hashtags for constitency and readability of the configuration.

## [0.5] - 2015-11-13
### Added
- do_not_retweet_hashtags parameter for not retweeting if a hashtag is in the text of the tweet.

## [0.4] - 2015-10-25
### Added
- use sqlite database through sqlalchemy to store already sent tweet ids.

## [0.3] - 2015-09-08
### Added
- retweets parameter to retweet only if a tweet already has been retweeted the number of times indicated by this parameter.
- waitminsecs parameter for the minimal seconds to wait before processing the next tweets.
- waitmaxsecs parameter for the maximal seconds to wait before processing the next tweets.

## [0.3] - 2015-09-08
### Added
- retweets parameter to retweet only if a tweet already has been retweeted the number of times indicated by this parameter.
- waitminsecs parameter for the minimal seconds to wait before processing the next tweets.
- waitmaxsecs parameter for the maximal seconds to wait before processing the next tweets.

## [0.2] - 2015-08-25
### Added
- bump Python version to 3.4.
- improve pep8 compliance.

## [0.1] - 2015-07-30
### Added
- initial version.


retweet/confparse.py
# -*- coding: utf-8 -*-
# Copyright © 2015 Carl Chenet <carl.chenet@ohmytux.com>
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/

# Get values of the configuration file
'''Get values of the configuration file'''

# standard library imports
import configparser
import sys

class ConfParse(object):
    '''ConfParse class'''
    def __init__(self, pathtoconf):
        '''Constructor of the ConfParse class'''
        self.user_to_retweet = 'journaldupirate'
        self.consumer_key = ''
        self.consumer_secret = ''
        access_token = ''
        self.access_token_secret = ''
        self.pathtoconf = pathtoconf
        self.dontretweethashes = []
        self.onlyiftags = []
        self.olderthan = 0
        self.youngerthan = 0
        self.main()

    def main(self):
        '''Main of the ConfParse class'''
        # read the configuration file
        config = configparser.ConfigParser()
        try:
            with open(self.pathtoconf) as conffile:
                config.read_file(conffile)
                if config.has_section('main'):
                    self.user_to_retweet = config.get('main', 'screen_name_of_the_user_to_retweet')
                    self.consumer_key = config.get('main', 'consumer_key')
                    self.consumer_secret = config.get('main', 'consumer_secret')
                    self.access_token = config.get('main', 'access_token')
                    self.access_token_secret = config.get('main', 'access_token_secret')
                    self.retweets = config.get('main', 'retweets')
                    self.waitminsecs = config.get('main', 'waitminsecs')
                    self.waitmaxsecs = config.get('main', 'waitmaxsecs')
                    if config.has_option('main', 'do_not_retweet_hashes'):
                        dontretweethashes = config.get('main', 'do_not_retweet_hashtags')
                        if dontretweethashes:
                            hashes = [i for i in dontretweethashes.split(',') if i != '']
                            self.dontretweethashes = hashes
                    if config.has_option('main', 'only_if_hashtags'):
                        onlyiftags = config.get('main', 'only_if_hashtags')
                        if onlyiftags:
                            hashtags = [i for i in onlyiftags.split(',') if i != '']
                            self.onlyiftags = hashtags
                    if config.has_option('main', 'older_than'):
                        self.olderthan = config.get('main', 'older_than')
                    if config.has_option('main', 'younger_than'):
                        self.youngerthan = config.get('main', 'younger_than')
                if config.has_section('sqlite'):
                    self.sqlitepath = config.get('sqlite', 'sqlitepath')

        except (configparser.Error, IOError, OSError) as err:
            print(err)
            sys.exit(1)
        try:
            self.retweets = int(self.retweets)
        except ValueError as err:
            print(err)
            self.retweets = 0
        try:
            self.waitminsecs = int(self.waitminsecs)
        except ValueError as err:
            print(err)
            self.waitminsecs = 60
        try:
            self.waitmaxsecs = int(self.waitmaxsecs)
        except ValueError as err:
            print(err)
            self.waitmaxsecs = 600
        try:
            self.olderthan = int(self.olderthan)
        except ValueError as err:
            print(err)
            self.olderthan = 0
        try:
            self.youngerthan = int(self.youngerthan)
        except ValueError as err:
            print(err)
            self.youngerthan = 0

    @property
    def confvalues(self):
        '''get the values of the configuration file'''
        return {'user_to_retweet':  self.user_to_retweet,
                'consumer_key': self.consumer_key,
                'consumer_secret': self.consumer_secret,
                'access_token': self.access_token,
                'access_token_secret': self.access_token_secret,
                'retweets': self.retweets,
                'waitminsecs': self.waitminsecs,
                'waitmaxsecs': self.waitmaxsecs,
                'sqlitepath': self.sqlitepath,
                'dontretweethashes': self.dontretweethashes,
                'onlyifhashtags': self.onlyiftags,
                'olderthan': self.olderthan,
                'youngerthan': self.youngerthan}
